src/Command/Ingester/LtiInstanceIngesterCommand.php

```diff
@@ -100,11 +102,15 @@ protected function execute(InputInterface $input, OutputInterface $output): int
             $this->progressBar->start();
 
             $numberOfProcessedRows = 0;
+            $currentTime = Carbon::now();
             foreach ($this->csvReader->getRecords() as $rawLtiInstance) {
                 $this->validateRow($rawLtiInstance, 'label', 'ltiLink', 'ltiKey', 'ltiSecret');
 
                 $numberOfProcessedRows++;
-                $this->ltiInstanceRepository->persist($this->createLtiInstance($rawLtiInstance));
+                $this->ltiInstanceRepository->persist($this->createLtiInstance($rawLtiInstance, $currentTime));
+
+                // It's crucial to not have identical creation time of LTI instances because of sorting reasons.
+                $currentTime->addMicroseconds(1);
```

Comment:
This seems to be very strange for me. Maybe we could consider to use this https://uuid.ramsey.dev/en/latest/customize/ordered-time-codec.html#customize-ordered-time-codec.

Reply:
This is a good suggestion 👍  but I have to concerns here, let's discuss:

1, UUID version 1 is time based indeed, but it also uses the host machine's MAC address to generate the ID. My question is, is it a good idea to expose this information (We'll use this IDs as resource IDs most probably in API)

2, Based on my testing (but please feel free to test it for yourself just to be sure) going with UUIDv1 does not solve the problem of ingesting thousands on resources in batches (it's most visible with users and assignments), because there is a chance of ID collision if they are generated in the exact same microsecond within a for loop. I tested it with maximum possible precision (microsecond accuracy) and there was always a few rows which were inserted in the exact same microsecond. Personally I couldn't find a better approach than artificially controlling the current time.

What do you think?

Reply:
Ok based on our discussion I am going to test alternatives